philipmeneghini/InstaConnect
# Lazy Loading For Profile Page Posts

**Summary**
This PR incorporates lazy loading into how posts are loaded on profile pages. Currently the application loads all posts at once. This will not be feasible if the user has thousands of posts. In order to do this I created a new `useLazyContents` hook which will dynamically fetch new posts for the profile as the user scrolls down.

![LazyLoadingProfilePosts](https://github.com/user-attachments/assets/acffa91c-07b0-459d-8964-c77ade1f98f2)

**Testing**

- Verify that all posts will eventually be loaded
- Check that posts will only load in specific increments
- Make sure that the posts endpoints are not called excessively
- Check for normal functionality on the profile page
